autorest/responder.go

```diff
@@ -97,6 +97,90 @@ func ByClosingIfError() RespondDecorator {
 	}
 }
 
+// ByUnmarshallingBool returns a RespondDecorator that decodes the http.Response Body into a bool
+// pointed to by b.
+func ByUnmarshallingBool(b *bool) RespondDecorator {
```

Comment:
Swagger expressly allows for primitive return types. Go's JSON decoder does not correctly handle those cases. We need to explicitly catch and convert using these responders.